packages/components/number-field/src/number-field.ts

```diff
@@ -108,23 +111,21 @@ export class NumberField extends LocaleMixin(TextField) {
       this.input.min = this.min.toString();
     }
 
-    if (changes.has('locale') || changes.has('formatOptions')) {
+    if (changes.has('formatOptions') || changes.has('locale')) {
```

Comment:
When I open the story
https://kind-plant-088739303-1840.westeurope.1.azurestaticapps.net/?path=/story/form-number-field--format-currency
and there is a value like `€9.90` in the number field:

<img width="320" alt="image" src="https://github.com/user-attachments/assets/0e5e3f7b-3964-4f30-bb6c-c20f9f38a094" />

and I change the locale in options e.g. to `de` the value is being removed from the number field 
and when I type in a value, the currency symbol is not being added, the value is not being formatted there and the value is invalid (on `main` it works fine)
<img width="1343" alt="image" src="https://github.com/user-attachments/assets/5401312e-41a6-4b44-b95e-867f22add0e9" />

Reply:
Also with the percentage, when I type in sth the value is not being formatted and the `%` is not added there
https://kind-plant-088739303-1840.westeurope.1.azurestaticapps.net/?path=/story/form-number-field--format-percent&args=locale:de

Reply:
same with unit
https://kind-plant-088739303-1840.westeurope.1.azurestaticapps.net/?path=/story/form-number-field--format-unit&args=locale:de